common/src/TaskCompletionSource.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    internal sealed class TaskCompletionSource : TaskCompletionSource<int>
```

Comment:
TaskCompletionSource is natively supported in NET 5.0. Our sdk used a Dotnetty's implementation for lower versions of the framework, which we have now brought into our sdk directly. The implementation is pretty straight-forward.

Reply:
Sounds like a worthy doc comment.